src/main/php/lang/ast/syntax/PHP.class.php

```diff
@@ -1045,6 +1051,30 @@ private function properties($parse, &$body, $meta, $modifiers, $type, $holder) {
     $parse->expecting(';', 'field declaration');
   }
 
+  /** Parses PHP8-style attributes (@@test) */
+  public function attribute($parse, &$annotations, $context) {
+    $name= $parse->token->value;
+    $parse->forward();
+
+    if ('(' === $parse->token->value) {
+      $parse->expecting('(', 'annotations');
+      $arguments= $this->expressions($parse);
+      if (empty($arguments)) {
+        $value= null;
+      } else if (1 === sizeof($arguments)) {
+        $value= $arguments[0];
+      } else {
+        $value= $arguments;
+      }
+      $parse->expecting(')', 'annotations');
```

Comment:
This should be re-considered - and maybe moved to either emitter or to XP Framework...

Reply:
We could change annotation layout from *an optional value* to *a list arguments*, giving us:

```php
// No argument / without value
@@Test #[@test]
$arguments= [];

// Single argument / with value
@@Author('Timm') #[@author('Timm')]
$arguments= ['Timm'];

// Single array argument / array value
@@Values(['a', 'b', 'c']) #[@values(['a', 'b', 'c')]
$arguments= [['a', 'b', 'c']];

// Multiple arguments - not supported by XP annotations!
@@Values('a', 'b', 'c')
$arguments= ['a', 'b', 'c']
```

This would create a BC break for this library